File: lesson_5_1/app/src/main/java/com/kalashnyk/denys/simplemvpapp/presentation/adapter/UserAdapter.java
package com.kalashnyk.denys.simplemvpapp.presentation.adapter;

import android.content.Context;
import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import com.kalashnyk.denys.simplemvpapp.R;
import com.kalashnyk.denys.simplemvpapp.data.database.entity.UserEntity;
import com.kalashnyk.denys.simplemvpapp.presentation.base.BaseAdapter;
import com.kalashnyk.denys.simplemvpapp.presentation.item.IUserItemClickListener;
import com.kalashnyk.denys.simplemvpapp.presentation.item.UserViewHolder;

import java.util.List;

public class UserAdapter extends BaseAdapter<UserViewHolder, UserEntity, IUserItemClickListener> {

    private List<UserEntity> mUsers;
    private Context mContext;

    public UserAdapter(Context context, List<UserEntity> list) {
        super(context, list);
        mUsers = list;
        mContext = context;
    }

    @NonNull
    @Override
    public UserViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new UserViewHolder(LayoutInflater.from(mContext).inflate(R.layout.item_user, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull UserViewHolder userViewHolder, int i) {
        super.onBindViewHolder(userViewHolder, i);
        userViewHolder.bind(mUsers.get(i), mItemClickListener);
    }

    @Override
    public int getItemCount() {
        return mUsers.size();
    }
}

File: lesson_9/app/src/main/java/com/kalashnyk/denys/kotlinbindingsample/presentation/adapter/UserAdapter.kt
package com.kalashnyk.denys.kotlinbindingsample.presentation.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import com.kalashnyk.denys.kotlinbindingsample.databinding.UserItemBinding
import com.kalashnyk.denys.kotlinbindingsample.presentation.base.BaseAdapter
import com.kalashnyk.denys.kotlinbindingsample.presentation.item.IUserItemClickListener
import com.kalashnyk.denys.kotlinbindingsample.presentation.item.UserViewHolder
import com.kalashnyk.denys.kotlinbindingsample.repository.database.entity.UserEntity

class UserAdapter(private val context: Context, private val users: List<UserEntity>, private val listener: IUserItemClickListener<UserEntity>) :
    BaseAdapter<UserViewHolder, UserEntity, IUserItemClickListener<UserEntity>>(users as MutableList<UserEntity>, listener) {

    override fun getItemCount(): Int {
        return users.size
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): UserViewHolder {
        var inflater: LayoutInflater = LayoutInflater.from(context)
        var binding: UserItemBinding = UserItemBinding.inflate(inflater, parent, false)
        return UserViewHolder(binding)
    }

    override fun onBindViewHolder(userViewHolder: UserViewHolder, i: Int) {
        super.onBindViewHolder(userViewHolder, i)
        userViewHolder.bind(users[i], listener)
    }
}
